README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# spectralem

[![R-CMD-check](https://github.com/jan-meissner/spectralem/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/jan-meissner/spectralem/actions/workflows/R-CMD-check.yaml)
[![codecov](https://codecov.io/github/jan-meissner/spectralem/branch/main/graphs/badge.svg)](https://app.codecov.io/gh/jan-meissner/spectralem/commits?branch=main)

## Installation

If `RcppFaddeeva` is not available for the current R version use:

``` r
remotes::install_github("cran/RcppFaddeeva")
```

You can install the development version with:

``` r
remotes::install_github("jan-meissner/spectralem")
```

## Examples

Fit Voigt profiles to some signal defined by `x` and `y`.

``` r
library(spectralem)

data <- synthetic.signal(seed = 1, K = 30, noise = 0.001)
x <- data$x
y <- data$y

res <- spectralem(x, y, K = 30)
#>   |                                                                              |                                                                      |   0%  |                                                                              |                                                                      |   1%  |                                                                              |=                                                                     |   1%  |                                                                              |=                                                                     |   2%  |                                                                              |==                                                                    |   2%  |                                                                              |==                                                                    |   3%  |                                                                              |===                                                                   |   4%  |                                                                              |===                                                                   |   5%  |                                                                              |====                                                                  |   5%  |                                                                              |====                                                                  |   6%  |                                                                              |=====                                                                 |   7%  |                                                                              |=====                                                                 |   8%  |                                                                              |======                                                                |   8%  |                                                                              |======                                                                |   9%  |                                                                              |=======                                                               |   9%  |                                                                              |=======                                                               |  10%  |                                                                              |=======                                                               |  11%  |                                                                              |========                                                              |  11%  |                                                                              |========                                                              |  12%  |                                                                              |=========                                                             |  12%  |                                                                              |=========                                                             |  13%  |                                                                              |==========                                                            |  14%  |                                                                              |==========                                                            |  15%  |                                                                              |===========                                                           |  15%  |                                                                              |===========                                                           |  16%  |                                                                              |============                                                          |  17%  |                                                                              |============                                                          |  18%  |                                                                              |=============                                                         |  18%  |                                                                              |=============                                                         |  19%  |                                                                              |==============                                                        |  19%  |                                                                              |==============                                                        |  20%  |                                                                              |==============                                                        |  21%  |                                                                              |===============                                                       |  21%  |                                                                              |===============                                                       |  22%  |                                                                              |================                                                      |  22%  |                                                                              |================                                                      |  23%  |                                                                              |=================                                                     |  24%  |                                                                              |=================                                                     |  25%  |                                                                              |==================                                                    |  25%  |                                                                              |==================                                                    |  26%  |                                                                              |===================                                                   |  27%  |                                                                              |===================                                                   |  28%  |                                                                              |====================                                                  |  28%  |                                                                              |====================                                                  |  29%  |                                                                              |=====================                                                 |  29%  |                                                                              |=====================                                                 |  30%  |                                                                              |=====================                                                 |  31%  |                                                                              |======================                                                |  31%  |                                                                              |======================                                                |  32%  |                                                                              |=======================                                               |  32%  |                                                                              |=======================                                               |  33%  |                                                                              |========================                                              |  34%  |                                                                              |========================                                              |  35%  |                                                                              |=========================                                             |  35%  |                                                                              |=========================                                             |  36%  |                                                                              |==========================                                            |  37%  |                                                                              |==========================                                            |  38%  |                                                                              |===========================                                           |  38%  |                                                                              |===========================                                           |  39%  |                                                                              |============================                                          |  39%  |                                                                              |============================                                          |  40%  |                                                                              |============================                                          |  41%  |                                                                              |=============================                                         |  41%  |                                                                              |=============================                                         |  42%  |                                                                              |==============================                                        |  42%  |                                                                              |==============================                                        |  43%  |                                                                              |===============================                                       |  44%  |                                                                              |===============================                                       |  45%  |                                                                              |================================                                      |  45%  |                                                                              |================================                                      |  46%  |                                                                              |=================================                                     |  47%  |                                                                              |=================================                                     |  48%  |                                                                              |==================================                                    |  48%  |                                                                              |==================================                                    |  49%  |                                                                              |===================================                                   |  49%  |                                                                              |===================================                                   |  50%  |                                                                              |===================================                                   |  51%  |                                                                              |====================================                                  |  51%  |                                                                              |====================================                                  |  52%  |                                                                              |=====================================                                 |  52%  |                                                                              |=====================================                                 |  53%  |                                                                              |======================================                                |  54%  |                                                                              |======================================                                |  55%  |                                                                              |=======================================                               |  55%  |                                                                              |=======================================                               |  56%  |                                                                              |========================================                              |  57%  |                                                                              |========================================                              |  58%  |                                                                              |=========================================                             |  58%  |                                                                              |=========================================                             |  59%  |                                                                              |==========================================                            |  59%  |                                                                              |==========================================                            |  60%  |                                                                              |==========================================                            |  61%  |                                                                              |===========================================                           |  61%  |                                                                              |===========================================                           |  62%  |                                                                              |============================================                          |  62%  |                                                                              |============================================                          |  63%  |                                                                              |=============================================                         |  64%  |                                                                              |=============================================                         |  65%  |                                                                              |==============================================                        |  65%  |                                                                              |==============================================                        |  66%  |                                                                              |===============================================                       |  67%  |                                                                              |===============================================                       |  68%  |                                                                              |================================================                      |  68%  |                                                                              |================================================                      |  69%  |                                                                              |=================================================                     |  69%  |                                                                              |=================================================                     |  70%  |                                                                              |=================================================                     |  71%  |                                                                              |==================================================                    |  71%  |                                                                              |==================================================                    |  72%  |                                                                              |===================================================                   |  72%  |                                                                              |===================================================                   |  73%  |                                                                              |====================================================                  |  74%  |                                                                              |====================================================                  |  75%  |                                                                              |=====================================================                 |  75%  |                                                                              |=====================================================                 |  76%  |                                                                              |======================================================                |  77%  |                                                                              |======================================================                |  78%  |                                                                              |=======================================================               |  78%  |                                                                              |=======================================================               |  79%  |                                                                              |========================================================              |  79%  |                                                                              |========================================================              |  80%  |                                                                              |========================================================              |  81%  |                                                                              |=========================================================             |  81%  |                                                                              |=========================================================             |  82%  |                                                                              |==========================================================            |  82%  |                                                                              |==========================================================            |  83%  |                                                                              |===========================================================           |  84%  |                                                                              |===========================================================           |  85%  |                                                                              |============================================================          |  85%  |                                                                              |============================================================          |  86%  |                                                                              |=============================================================         |  87%  |                                                                              |=============================================================         |  88%  |                                                                              |==============================================================        |  88%  |                                                                              |==============================================================        |  89%  |                                                                              |===============================================================       |  89%  |                                                                              |===============================================================       |  90%  |                                                                              |===============================================================       |  91%  |                                                                              |================================================================      |  91%  |                                                                              |================================================================      |  92%  |                                                                              |=================================================================     |  92%  |                                                                              |=================================================================     |  93%  |                                                                              |==================================================================    |  94%  |                                                                              |==================================================================    |  95%  |                                                                              |===================================================================   |  95%  |                                                                              |===================================================================   |  96%  |                                                                              |====================================================================  |  97%  |                                                                              |====================================================================  |  98%  |                                                                              |===================================================================== |  98%  |                                                                              |===================================================================== |  99%  |                                                                              |======================================================================|  99%  |                                                                              |======================================================================| 100%

# get position of fitted peaks
res$fit_params$pos
#>  [1] 670.9607 503.3100 316.2073 274.8785 801.8549 251.4250 706.4645 411.1184
#>  [9] 772.3675 354.4219 187.9791 549.1562 854.1438 726.5067 326.5962 572.6050
#> [17] 314.2619 688.0916 502.1091 516.8439 203.5130 238.0623 810.7499 772.4268
#> [25] 253.0410 549.7478 308.1590 178.2623 801.5274 276.4495

# plot the fit
library(ggplot2)
pd <- data.frame(x = x, y = y, fit = voigt.model(x, res$fit_params))
ggplot(pd, aes(x)) +
  geom_line(aes(y = y, colour = "y")) +
  geom_line(aes(y = fit, colour = "fit")) +
  theme_bw()
```

<img src="man/figures/README-unnamed-chunk-2-1.svg" width="100%" />


NEWS.md
# spectralem 0.9

* Almost CRAN ready

# spectralem 0.1

* Added a `NEWS.md` file to track changes to the package.


tools/what.R


#####################
get_toluene_measured_data <- function(){
  path <- "C:\\Users\\Jan\\Desktop\\ISW_SpectraBayes\\ISW_SpectraBayes\\bayes_mcmc\\dev\\wip_gibbs\\golden_gibbs\\toluene_measured.csv"
  toluene_measured <- read.csv(path, header=FALSE, sep=";")
  x <- as.vector(toluene_measured[,1])
  ytrue <- as.vector(toluene_measured[,2])
  return(list(x=x,y=ytrue))
}

library(remotes)
remotes::install_version("RcppFaddeeva", "0.2.2")
library(RcppFaddeeva)
remotes::install_github("jan-meissner/spectralem")

library("spectralem")
data <- data.synthetic.stormyclouds(seed = 1, kp = 30, noise = 0.001)
x <- data$x
y <- data$y

res <- spectralem(x, y, max_peaks = 3, max_iter = 40)


install.packages('plotly')
library("plotly")
fig <- plotly::plot_ly(x = x, y = y, name = 'true', type = 'scatter', mode = 'lines')
fig <- fig %>% plotly::add_lines(y = res$fit, name = 'fit', type = 'scatter', mode = 'lines')
fig

#library(proftools)
#pd <- proftools::readProfileData("profile.out")
#head(funSummary(pd), 10)


library(babynames) # provide the dataset: a dataframe called babynames
library(dplyr)

# Keep only 3 names
don <- babynames %>%
filter(name %in% c("Ashley", "Patricia", "Helen")) %>%
filter(sex=="F")

library(ggplot2)
pd <- data.frame(y = c(res$fit, y), x=rep(x, 2), type = rep(c("fit","true"), each = length(x)))
pd %>% ggplot( aes(x=x, y=y, color=type)) + geom_line() + theme_bw()